### PR TITLE
TEST/IODEMO: support set multiple client ip addresses

### PR DESCRIPTION
**Pull request by changchengx**
Signed-off-by: Changcheng Liu <jerrliu@nvidia.com>

## What
Support set multiple client ip addresses.

## Why ?
Feature requirement

### CODE REVIEW

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -2074,8 +2075,11 @@ class DemoClient : public P2pDemoCommon {
             abort();
         }
 
-        if (opts().src_addr != NULL) {
-            ret = set_sockaddr(opts().src_addr, 0, (struct sockaddr*)&src_addr);
+        if (opts().src_addrs.size() != 0) {
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:2078]: !opts().src_addrs.empty()

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -2822,7 +2837,7 @@ static int parse_args(int argc, char **argv, options_t *test_opts)
             }
             break;
         case 'I':
-            test_opts->src_addr = optarg;
+            set_src_addrs(optarg, test_opts);
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:2840]: maybe -I could be passed multiple times. every -I argument adds another address. then, no need to parse by ,

File: test/apps/iodemo/io_demo.cc (hunk 1 above)
Comment (changchengx) [reply test/apps/iodemo/io_demo.cc]: Done

File: test/apps/iodemo/io_demo.cc (hunk 2 above)
Comment (changchengx) [reply test/apps/iodemo/io_demo.cc]: Good idea. This could simplify implementation and minimize change.

Comment (changchengx) [reply test/apps/iodemo/io_demo.cc]: Done

**Review by yosefe — APPROVED** [review]
please squash and port to branch

**Pull request merged**